tower-batch/src/service.rs

```diff
@@ -94,9 +94,11 @@ where
         // Then, poll to acquire a semaphore permit. If we acquire a permit,
         // then there's enough buffer capacity to send a new request. Otherwise,
         // we need to wait for capacity.
-        ready!(self.semaphore.poll_acquire(cx));
-
-        Poll::Ready(Ok(()))
+        let res = self.semaphore.poll_acquire(cx);
+        match ready!(res) {
+            Ok(()) => Poll::Ready(Ok(())),
+            _ => Poll::Ready(Err(self.get_worker_error())),
+        }
```

Comment:
Since the semaphore can't error, there's no need for us to map the error type here.

Instead, we should document the difference between tokio 0.3.7 and 1.0:
```suggestion
        // In tokio 0.3.7, `acquire_owned` panics if its semaphore returns an error,
        // so we don't need to handle errors until we upgrade to tokio 1.0.
        ready!(self.semaphore.poll_acquire(cx));

        Poll::Ready(Ok(()))
```